### PR TITLE
feat: Add BDD tests

### LINKED ISSUE #4: Initial BDD test
Initial server components: 
- initial CAS IPFS client
- core observer
- mocks : protocol, store and operation queue (to be replaced with DB-backed implementations)

Initial BDD test

### PR DESCRIPTION
**Pull request by sandrask**
Initial BDD tests

Closes #4

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

**Review by bstasyszyn — APPROVED** [review]

File: test/bdd/fixtures/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    ports:
+      - 48326:48326
+    volumes:
+      - ./keys/tls:/etc/orb/tls
+    depends_on:
+      - ipfs
+
+  ipfs:
+    container_name: ipfs
+    #this image doesn't work on ARM64 yet
```
Comment (troyronda) [inline test/bdd/fixtures/docker-compose.yml:32]: We should also have a TODO to support multi-arch docker images (amd64 + arm64).

File: ci/azure-pipelines.yml [hunk 2]
```diff
@@ -28,9 +28,20 @@ jobs:
         CODECOV_UPLOAD_TOKEN: $(CODECOV_UPLOAD_TOKEN)
       displayName: Upload coverage to Codecov
 
+  - job: BDDTest
+    pool:
+      vmImage: ubuntu-16.04
```
Comment (troyronda) [inline ci/azure-pipelines.yml:33]: Assume we can update this to the latest ubuntu LTS?

Comment (sandrask) [reply ci/azure-pipelines.yml]: Fixed

File: test/bdd/fixtures/docker-compose.yml (hunk 1 above)
Comment (sandrask) [reply test/bdd/fixtures/docker-compose.yml]: Added TODO

**Pull request merged**